crates/storage/trie.rs

```diff
@@ -34,27 +37,27 @@ pub type ValueRLP = Vec<u8>;
 
 /// Libmdx-based Ethereum Compatible Merkle Patricia Trie
 /// Adapted from https://github.com/lambdaclass/merkle_patricia_tree
-pub struct Trie {
+pub struct Trie<DB: TrieDB> {
     /// Hash of the current node
     root: Option<NodeHash>,
     /// Contains the trie's nodes
-    pub(crate) state: TrieState,
+    pub(crate) state: TrieState<DB>,
 }
 
-impl Trie {
+impl<DB: TrieDB> Trie<DB> {
     /// Creates a new Trie from a clean DB
-    pub fn new(trie_dir: &str) -> Result<Self, StoreError> {
+    pub fn new(db: DB) -> Result<Self, StoreError> {
         Ok(Self {
-            state: TrieState::create(trie_dir)?,
+            state: TrieState::new(db),
             root: None,
         })
     }
```

Comment:
We should remove the Result if the method cannot fail

Reply:
You're right! Removed it.